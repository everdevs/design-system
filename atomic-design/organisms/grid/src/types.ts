import React from "react";
import { GridConfig, PropsWithTheme } from "@evernest/theme";

export type ColSpanValue = string | number;

export interface ColSpan {
	xs?: ColSpanValue;
	s?: ColSpanValue;
	m?: ColSpanValue;
	l?: ColSpanValue;
}

export interface ColumnProps extends ColSpan, PropsWithTheme {
	raw?: boolean;
	flex?: boolean;
	order?: number;
}
export type ColumnType = React.ReactElement<ColumnProps> | React.ReactElement<ColumnProps>[];

export enum Alignment {
	start = "flex-start",
	end = "flex-end",
	center = "center",
}

export interface RowProps extends PropsWithTheme {
	children?: ColumnType;
	raw?: boolean;
	noWrap?: boolean;
	reverse?: boolean;
	justify?: Alignment;
}

export type RowType = React.ReactElement<RowProps> | React.ReactElement<RowProps>[];

export interface GridProps extends PropsWithTheme {
	children?: RowType;
	overflow?: boolean;
}

export type GridOverlayPosition = "fixed" | "absolute";

export interface GridOverlayGridProps extends GridProps {
	position?: GridOverlayPosition;
}

export interface GridOverlayProps extends GridOverlayGridProps {
	initialActive?: boolean;
	toggle?: boolean;
}

export interface ViewportSize {
	m: boolean;
	mu: boolean;
	l: boolean;
	lu: boolean;
	xl: boolean;
}

export interface GridContextConfig extends GridConfig {
	viewport: ViewportSize;
}

export interface GridContextProps {
	grid: GridConfig;
}

export interface BoxProps {
	removePadding?: boolean;
	addPadding?: boolean;
	removeGutter?: boolean;
	addGutter?: boolean;
	flex?: boolean;
}
