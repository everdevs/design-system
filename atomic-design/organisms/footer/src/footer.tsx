import { css } from "@emotion/core";
import styled from "@emotion/styled";
import React from "react";
import { StyledFooterProps, FooterProps } from "./types";

export const StyledFooter = styled.div<StyledFooterProps>`
	padding: var(--spacing-xl) 0;
	${({ theme: { palette } }) => css`
		background: ${palette.darkGrey.css};
		color: ${palette.white.css};
	`};
`;

export const Footer = React.forwardRef<FooterProps, StyledFooterProps>((props, ref) => (
	<StyledFooter {...props} ref={ref} />
));
