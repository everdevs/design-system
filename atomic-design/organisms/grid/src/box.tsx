import { css } from "@emotion/react";
import styled from "@emotion/styled";
import React from "react";
import { GUTTER, PADDING } from "./constants";
import { BoxProps } from "./types";

const StyledBox = styled.div<BoxProps>`
	flex: 1;
	${({ flex }) => css`
		display: ${flex ? "flex" : "block"};
	`};
	${({ removePadding, removeGutter, addGutter, addPadding }) => {
		if (removePadding) {
			if (removeGutter) {
				return css`
					margin-left: calc((var(${PADDING}) + var(${GUTTER})) * -1px);
					margin-right: calc((var(${PADDING}) + var(${GUTTER})) * -1px);
				`;
			}
			return css`
				margin-left: calc(var(${PADDING}) * -1px);
				margin-right: calc(var(${PADDING}) * -1px);
			`;
		}
		if (removeGutter) {
			return css`
				margin-left: calc(var(${GUTTER}) * -1px);
				margin-right: calc(var(${GUTTER}) * -1px);
			`;
		}
		if (addGutter) {
			if (addPadding) {
				return css`
					padding-left: calc(var(${PADDING}) * 1px);
					padding-right: calc(var(${PADDING}) * 1px);
					margin-left: calc(var(${GUTTER}) * 1px);
					margin-right: calc(var(${GUTTER}) * 1px);
				`;
			}
			return css`
				margin-left: calc(var(${GUTTER}) * 1px);
				margin-right: calc(var(${GUTTER}) * 1px);
			`;
		}
		return css`
			padding-left: calc(var(${PADDING}) * 1px);
			padding-right: calc(var(${PADDING}) * 1px);
		`;
	}};
`;

export const Box: React.FC<BoxProps> = props => {
	return <StyledBox {...props} />;
};
