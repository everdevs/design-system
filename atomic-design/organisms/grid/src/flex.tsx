import styled from "@emotion/styled";

export const Flex = styled.div`
	display: flex;
	justify-content: space-between;
	align-content: center;
	align-items: center;
`;
