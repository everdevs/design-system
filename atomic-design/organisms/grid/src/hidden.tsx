import styled from "@emotion/styled";
import { PropsWithTheme } from "@evernest/theme";
import { css } from "@emotion/react";

export interface HiddenProps extends PropsWithTheme {
	display?: string;
	s?: boolean;
	m?: boolean;
	l?: boolean;
}
export const Hidden = styled.div<HiddenProps>`
	${({ theme: { mq }, display, s, m, l }) => css`
		display: ${s ? display : "none"};
		@media ${mq.m} {
			display: ${m ? display : "none"};
		}

		@media ${mq.l} {
			display: ${l ? display : "none"};
		}
	`};
`;

Hidden.defaultProps = {
	display: "block",
};
