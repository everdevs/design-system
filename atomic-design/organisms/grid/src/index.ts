export * from "./box";
export * from "./column";
export * from "./constants";
export * from "./flex";
export * from "./hidden";
export * from "./grid";
export * from "./grid-overlay";
export * from "./row";
export * from "./stage";
export * from "./types";
