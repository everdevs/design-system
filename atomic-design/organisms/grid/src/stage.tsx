import { COLCOUNT, COLSPAN } from "./constants";
import styled from "@emotion/styled";

export const Stage = styled.div`
	${COLCOUNT}: var(${COLSPAN});
	display: flex;
	width: 100vw;
	margin-left: calc((-100vw + 100%) / 2);
	margin-right: calc((-100vw + 100%) / 2);
`;
