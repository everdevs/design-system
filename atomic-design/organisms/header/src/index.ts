export * from "./header";
export * from "./types";
export * from "./utils";
