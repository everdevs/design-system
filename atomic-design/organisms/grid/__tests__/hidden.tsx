import React from "react";
import { mount } from "enzyme";
import { Hidden, HiddenProps } from "../src";
import { createTree, withTheme } from "@evernest/dev-helpers";
import { theme } from "@evernest/theme";

const ThemedHidden = withTheme<HiddenProps>(Hidden, theme);
test("Hidden renders children", () => {
	const text = "Hello Hidden";
	const wrapper = mount(<ThemedHidden>{text}</ThemedHidden>);
	expect(wrapper.text()).toEqual(text);
});

test("Hidden allows setting classname", () => {
	const className = "test";
	const wrapper = mount(<ThemedHidden className={className} />);
	expect(wrapper).toHaveClassName(className);
});

test("Hidden can show/hide content on different viewports", () => {
	const s = createTree(<ThemedHidden s />);
	expect(s).toMatchSnapshot();
	const m = createTree(<ThemedHidden s />);
	expect(m).toMatchSnapshot();
	const l = createTree(<ThemedHidden s />);
	expect(l).toMatchSnapshot();
	const sm = createTree(<ThemedHidden s m />);
	expect(sm).toMatchSnapshot();
	const sl = createTree(<ThemedHidden s l />);
	expect(sl).toMatchSnapshot();
	const ml = createTree(<ThemedHidden m l />);
	expect(ml).toMatchSnapshot();
});
