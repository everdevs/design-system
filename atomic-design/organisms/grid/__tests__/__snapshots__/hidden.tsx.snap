// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hidden can show/hide content on different viewports 1`] = `
<div
  className="css-1kvm6ra"
  display="block"
/>
`;

exports[`Hidden can show/hide content on different viewports 2`] = `
<div
  className="css-1kvm6ra"
  display="block"
/>
`;

exports[`Hidden can show/hide content on different viewports 3`] = `
<div
  className="css-1kvm6ra"
  display="block"
/>
`;

exports[`Hidden can show/hide content on different viewports 4`] = `
<div
  className="css-a383ft"
  display="block"
/>
`;

exports[`Hidden can show/hide content on different viewports 5`] = `
<div
  className="css-zqakw9"
  display="block"
/>
`;

exports[`Hidden can show/hide content on different viewports 6`] = `
<div
  className="css-10ekhzz"
  display="block"
/>
`;
